src/antares/service/local_services/binding_constraint_local.py

```diff
@@ -47,6 +49,10 @@ def create_binding_constraint(
         equal_term_matrix: Optional[pd.DataFrame] = None,
         greater_term_matrix: Optional[pd.DataFrame] = None,
     ) -> BindingConstraint:
+        if transform_name_to_id(name) in self.binding_constraints:
```

Comment:
Mmmmh you assume the id equals transform_name_to_id(name) but it's an implementation detail of the class BindingConstraint. You should just put this code after creating the constraint as you can use the created id without assuming anything